src/Python/qsharp-core/qsharp/clients/iqsharp.py

```diff
@@ -127,6 +127,25 @@ def get_workspace_operations(self) -> List[str]:
     def reload(self) -> None:
         return self._execute(f"%workspace reload", raise_on_stderr=True)
 
+    def get_config(self) -> Dict[str, object]:
+        raw = self._execute(f"%config", raise_on_stderr=True)
+        config_settings = {}
+        for row in raw["rows"]:
+            config_settings[row['Key']] = row['Value']
+        return config_settings
+
+    def set_config(self, name : str, value : object) -> None:
+        from numbers import Number
+        if isinstance(value, bool):
+            self._execute(f"%config {name}={'true' if value else 'false'}", raise_on_stderr=True)
+        elif isinstance(value, Number):
+            self._execute(f"%config {name}={value}", raise_on_stderr=True)
+        else:
+            self._execute(f"%config {name}='{value}'", raise_on_stderr=True)
```

Comment:
We may want to consolidate this with existing support for preparing values for JSON serialization (`map_tuples` and `unmap_tuples`).

Reply:
Yes, actually today `%config` doesn't support JSON-style arguments. It needs to be updated to share the argument-parsing code with `%simulate`, etc., so that we can pass arguments in JSON format as with the other magic commands. I've opened #323 to track this.